build/BannedSymbols.txt

```diff
@@ -21,3 +21,9 @@ M:System.IO.Path.GetTempPath(); Use NuGetEnvironment.GetFolderPath(NuGetFolderPa
 
 M:NuGet.Protocol.Core.Types.SourceRepository.GetResource`1(); Pass a cancellation token
 M:NuGet.Protocol.Core.Types.SourceRepository.GetResourceAsync`1(); Pass a cancellation token
+
+M:System.Environment.GetEnvironmentVariable(System.String); Use NuGet.Common.IEnvironmentVariableReader
+M:System.Environment.GetEnvironmentVariable(System.String,System.EnvironmentVariableTarget); Use NuGet.Common.IEnvironmentVariableReader
+
+M:System.Environment.SetEnvironmentVariable(System.String,System.String); NuGet should not need to set environment variables. If you need to set one, you must seek an exception.
```

Comment:
nit: just curious if we should mention to find these guidelines in our repo, or even provide a link to the markdown?

Reply:
I'll add a link to the message